js/ui/grid_core/ui.grid_core.search.js

```diff
@@ -343,16 +343,30 @@ module.exports = {
                 },
 
                 _updateCell: function($cell, parameters) {
-                    var that = this,
-                        column = parameters.column,
+                    var column = parameters.column,
                         dataType = column.lookup && column.lookup.dataType || column.dataType,
                         isEquals = dataType !== "string";
 
                     if(allowSearch(column)) {
-                        that._highlightSearchText($cell, isEquals, column);
+                        if(this.option("templatesRenderAsynchronously")) {
+                            if(!this._searchParams) {
```

Comment:
It is more good way to initialize this._searchParams in init method
`init: function() {
    this.callBase.apply(this, arguments);
    this._searchParams = [];
}`